File: mvn-advent-of-code/src/test/java/advent/twenty_seventeen/Day09StreamProcessingTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day09StreamProcessingTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day09StreamProcessing("/twenty_seventeen/Day09-StreamProcessing-input.txt");

        assertEquals(10616, dp.getPart1Answer().intValue());
        assertEquals(5101, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day09StreamProcessing("/twenty_seventeen/Day09-StreamProcessing-example#1.txt");

        assertEquals(50, dp.getPart1Answer().intValue());
        assertEquals(29, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day09StreamProcessing("/twenty_seventeen/Day09-StreamProcessing-example#2.txt");

        assertEquals(0, dp.getPart1Answer().intValue());
        assertEquals(32, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day09StreamProcessingTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day09StreamProcessingTest  extends FlatSpec with Matchers {
  "2017 Day 09 - Input" should "calculate answers" in {
    val dp = new Day09StreamProcessing("/twenty_seventeen/Day09-StreamProcessing-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day09StreamProcessing("/twenty_seventeen/Day09-StreamProcessing-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
